frontend/app/work_packages/controllers/work-package-new-controller.js

```diff
@@ -0,0 +1,172 @@
+        vm.groupedFields = WorkPackagesOverviewService.getGroupedWorkPackageOverviewAttributes();
+        var schema = WorkPackageFieldService.getSchema(vm.workPackage);
+        var otherGroup = _.find(vm.groupedFields, { groupName: 'other' });
+        otherGroup.attributes = [];
+        _.forEach(schema.props, function(prop, propName) {
+          if (propName.match(/^customField/)) {
+            otherGroup.attributes.push(propName);
+          }
+        });
+        otherGroup.attributes.sort(function(a, b) {
+          return getLabel(a).toLowerCase().localeCompare(getLabel(b).toLowerCase());
+        });
+        if (!firstTimeFocused) {
+          firstTimeFocused = true;
+          $timeout(function() {
+            // TODO: figure out a better way to fix the wp table columns bug
+            // where arrows are misplaced when not resizing the window
+            angular.element($window).trigger('resize');
+            angular.element('.work-packages--details--subject .focus-input').focus();
+          });
+        }
+
+      });
+
+    });
+
+    $scope.$on('workPackageUpdatedInEditor', function(e, workPackage) {
+      $state.go('work-packages.list.details.overview', {workPackageId: workPackage.props.id});
+    });
+  }
+
+  function submit(notify) {
+    angular
+      .element('.work-packages--details--subject:first .inplace-edit--write')
+      .scope().editPaneController.submit(notify);
+  }
+
+  function cancel() {
+    // previousState set in a $stateChangeSuccess callback
+    // in the .run() sequence
+    if ($rootScope.previousState && $rootScope.previousState.name) {
+      vm.loaderPromise = $state.go($rootScope.previousState.name, $rootScope.previousState.params);
+    } else {
+      vm.loaderPromise = $state.go('^');
+    }
+  }
+
+  function isGroupHideable(groupName) {
+    var group = _.find(vm.groupedFields, {groupName: groupName});
+    return _.every(group.attributes, isFieldHideable);
+  }
+
+  function isFieldHideable(field) {
+    if (!isSpecified(field)) {
+      return true;
+    }
+
+    if (!isEditable(field)) {
+      return true;
+    }
+
+    if (_.contains(unhideableFields, field)) {
+      return !WorkPackageFieldService.isEditable(vm.workPackage, field);
+    }
+
+    if (WorkPackageFieldService.isRequired(vm.workPackage, field)) {
+      return false;
+    }
+    return WorkPackageFieldService.isHideable(vm.workPackage, field);
+  }
+
+  function isSpecified(field) {
+    return WorkPackageFieldService.isSpecified(vm.workPackage, field);
+  }
+
+  function isEditable(field) {
+    return WorkPackageFieldService.isEditable(vm.workPackage, field);
+  }
+
+  function hasNiceStar(field) {
+    return WorkPackageFieldService.isRequired(vm.workPackage, field) &&
+      WorkPackageFieldService.isEditable(vm.workPackage, field);
+  }
+
+  function getLabel(field) {
+    return WorkPackageFieldService.getLabel(vm.workPackage, field);
+  }
+
+  function showToggleButton() {
+    return true;
+  }
+};
```

Comment:
I can understand that there are reasons where you rather copy parts of a method instead of doing the business involved in DRYing it up. (read: Sometimes, under limited circumstances; don't read: best practice)

However, I think there are no reasons to copy over a bunch of helper methods (`isXXX`, `hasXXX`, etc.) from another class that are all completely identical. Here you have several options, among them:

* inheritance from a common base class (if that is a thing in JS)
* using a shared class offering helper methods